src/apps/lwaftr/icmp.lua

```diff
@@ -93,6 +95,22 @@ function new_icmpv4_packet(from_eth, to_eth, from_ip, to_ip, initial_pkt, l2_siz
    return new_pkt
 end
 
+function is_icmpv4(pkt)
+   return is_ipv4(pkt)
+      and pkt.data[constants.ethernet_header_size + constants.o_ipv4_proto] == constants.proto_icmp
```

Comment:
I agree with Andy here, but we need to actually properly *benchmark* this kind of stuff when we have the tooling for that set up. Until then, doing what he asks won't hurt speed and might help, at a minimal cost to how the code looks.